admin/horace_on.m.template

```diff
@@ -30,27 +41,36 @@ if ~isempty(sw_start)
     spinw_on(default_spinw_path);
 end
 
+if nargin > 1
+    independent_herbert = true;
+else
+    independent_herbert = false;
+end
```

Comment:
Why not 
```MATLAB
independent_herbert = exist('non_default_her_path', 'var') && ~isempty(non_default_her_path);
```
(or equivalent with if) which both makes intent clear and protects against later modification.

Reply:
I prefer avoiding checking for input variable existence, as in fact it only checks if the position of the variable is filled. 
I.e. if you have `my_fun(a,b)` and call it `my_fun(b)` the check inside `my_fun`  `exist('b','var')` would return `false`. So the check gives you false feeling that you are checking for variable existence when in fact you are checking for `nargin`. `isempty` can be added -- no harm.

Reply:
This reasoning doesn't make sense to me. You are using positional arguments, so of course it only checks the position. Within the function the name of the outer variable doesn't mean anything. You are checking exactly what you asked for. If you want to use keyword arguments, then you can use keyword arguments within `inputParser` and subsequently check for presence.

```matlab
p = inputParser();
addRequired(p,'a');
addParameter(p,'b', []);
parse(p,a, b);

if p.Results.usingDefaults.b
   ...
end
```
But there are complaints that this is slow in performance critical functions (while I don't think this is).

Checking nargin adds a level of abstraction in determining which argument is present or not, reduces extensibility and maintainability for other programmers and in fact, with newer (2022a) MATLAB keyword syntax what you are checking is fundamentally wrong.
```matlab
my_fun(b='b')
% nargin -> 1
% b -> exists
```